analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/InitializationVectorShouldBeRandomBase.cs

```diff
@@ -18,10 +18,108 @@
+
 namespace SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks;
 
 public abstract class InitializationVectorShouldBeRandomBase : SymbolicRuleCheck
 {
     protected const string DiagnosticId = "S3329";
     protected const string MessageFormat = "Use a dynamically-generated, random IV.";
+
+    protected override ProgramState PreProcessSimple(SymbolicContext context)
+    {
+        var state = context.State;
+        if (context.Operation.Instance.AsArrayCreation() is { } arrayCreation
+            && arrayCreation.Type.Is(KnownType.System_Byte_Array)
+            && arrayCreation.DimensionSizes.Length == 1)
+        {
+            return state.SetOperationConstraint(arrayCreation, ByteCollectionConstraint.CryptographicallyWeak);
+        }
+        else if (context.Operation.Instance.AsAssignment() is { } assignment)
+        {
+            return ProcessAssignmentToIVProperty(assignment, state)
+                   ?? state;
+        }
+        else if (context.Operation.Instance.AsInvocation() is { } invocation)
+        {
+            return ProcessStrongRandomGeneratorMethodInvocation(invocation, state)
+                    ?? ProcessCreateEncryptorMethodInvocation(invocation, state, context)
+                    ?? state;
+        }
+        else
+        {
+            return state;
+        }
+    }
+
+    private static ProgramState ProcessAssignmentToIVProperty(IAssignmentOperationWrapper assignment, ProgramState state)
+    {
+        return assignment.Target?.AsPropertyReference() is { } property
+                && GetIVPropertyInstance(property) is { } instanceSymbol
+                && (assignment.Value.AsArrayCreation() is { } || state[assignment.Value].HasConstraint(ByteCollectionConstraint.CryptographicallyWeak))
+                    ? state.SetSymbolConstraint(instanceSymbol, ByteCollectionConstraint.CryptographicallyWeak)
+                    : null;
+
+        static ISymbol GetIVPropertyInstance(IPropertyReferenceOperationWrapper property) =>
+            property.Property.Name.Equals("IV")
+            && property.Instance.TrackedSymbol() is { } instanceSymbol
+            && instanceSymbol.GetSymbolType().DerivesFrom(KnownType.System_Security_Cryptography_SymmetricAlgorithm)
+               ? instanceSymbol
+               : null;
+    }
+
+    private static ProgramState ProcessStrongRandomGeneratorMethodInvocation(IInvocationOperationWrapper invocation, ProgramState state)
+    {
+        if (IsCryptographicallyStrongRandomNumberGenerator(invocation)
+            && invocation.ArgumentValue("data") is { } byteArray
+            && byteArray.TrackedSymbol() is { } byteArraySymbol)
+        {
+            return state.SetSymbolConstraint(byteArraySymbol, ByteCollectionConstraint.CryptographicallyStrong);
+        }
+        else if (IsGenerateIVMethod(invocation))
+        {
+            return state.SetSymbolConstraint(invocation.Instance.TrackedSymbol(), ByteCollectionConstraint.CryptographicallyStrong);
```

Comment:
```suggestion
            return state.SetSymbolConstraint(state.ResolveCaptureAndUnwrapConversion(invocation.Instance).TrackedSymbol(), ByteCollectionConstraint.CryptographicallyStrong);
```

The more I use this, the more I think we need to handle this behavior in `TrackedSymbol`.

Reply:
The more I see it the more it makes sense indeed.

Reply:
It doesn't seem like you applied this.